src/highdicom/content.py

```diff
@@ -936,3 +1068,481 @@ def __init__(
+            raise TypeError('Argument "first_mapped_value" must be an integer.')
+        if first_mapped_value < 0:
+            raise ValueError(
+                'Argument "first_mapped_value" must be non-negative.'
+            )
+        if first_mapped_value >= 2 ** 16:
+            raise ValueError(
+                'Argument "first_mapped_value" must be less than 2^16.'
+            )
+
+        if not isinstance(lut_data, np.ndarray):
+            raise TypeError('Argument "lut_data" must be of type np.ndarray')
+        if lut_data.ndim != 1:
+            raise ValueError("Numpy array must have a single dimension.")
+        len_data = lut_data.size
+        if len_data == 0:
+            raise ValueError('Argument "lut_data" must not be empty.')
+        if len_data > 2**16:
+            raise ValueError(
+                'Length of lut_data must be no greater than 2^16 elements.'
+            )
+        elif len_data == 2**16:
+            # Per the standard, this is recorded as 0
+            len_data = 0
+        if lut_data.dtype.type == np.uint8:
+            bits_allocated = 8
+        elif lut_data.dtype.type == np.uint16:
+            bits_allocated = 16
+        else:
+            raise ValueError(
+                "Numpy array must have dtype uint8 or uint16."
+            )
+        self.LUTData = lut_data.tobytes()
+
+        self.LUTDescriptor = [
+            len_data,
+            int(first_mapped_value),
+            bits_allocated
+        ]
+
+        if lut_explanation is not None:
+            _check_long_string(lut_explanation)
+            self.LUTExplanation = lut_explanation
+
+    @property
+    def lut_data(self) -> np.ndarray:
+        """np.ndarray: LUT data formatted as np.ndarray."""
+        if self.bits_allocated == 8:
+            np_dtype = np.uint8
+        elif self.bits_allocated == 16:
+            np_dtype = np.uint16
+        else:
+            raise RuntimeError("Invalid LUT descriptor.")
+        lut_length = self.LUTDescriptor[0]
+        data = np.frombuffer(self.LUTData, np_dtype)
+        if len(data) != lut_length:
+            raise RuntimeError(
+                'Length of LUTData does not match the value expected from the '
+                f'LUTDescriptor. Expected {lut_length}, found {len(data)}.'
+            )
+        return data
+
+    @property
+    def first_mapped_value(self) -> int:
+        """int: Pixel value that will be mapped to the first value in the
+        LUT.
+        """
+        return int(self.LUTDescriptor[1])
+
+    @property
+    def bits_allocated(self) -> int:
+        """int: Bits allocated for the LUT data. 8 or 16."""
+        return int(self.LUTDescriptor[2])
+
+
+class ModalityLUT(LUT):
+
+    """Dataset describing an item of the Modality LUT Sequence."""
+
+    def __init__(
+        self,
+        lut_type: Union[RescaleTypeValues, str],
+        first_mapped_value: int,
+        lut_data: np.ndarray,
+        lut_explanation: Optional[str] = None
```

Comment:
See bb6129d110db6a9c3a1d20f2414a0f72fb4bdac6

Reply:
I think you were right about deriving `ModalityLUT` from `LUT`. After all, `ModalityLUT` and `VOILUT` are two very different beasts. I rolled back the changes to the implementation of `ModalityLUT` via 89e9422c5cc2b441a14769c853b405ef6b57bf91. I also added a comment to explain the differences between `ModalityLUT` and `VOILUT`.

Reply:
The confusion comes from the fact that there is a VOI LUT Sequence, which is very similar to a Modality LUT Sequence, but there are also Softcopy VOI LUT Sequence and Frame VOI LUT Sequence, which contain VOI LUT Sequence. That's a very unfortunate choice of terms.

Softcopy VOI LUT Sequence and Frame VOI LUT Sequence both include attributes of the VOI LUT Macro, which includes the VOI LUT Sequence.

The `ModalityLUT` class implements the Modality LUT Sequence, whereas the `VOILUT` class implements the VOI LUT **Macro**.

Shall we do something about that or just document it well?

Reply:
I added a warning to the docstring via 134ae1bc6f4ce4dd9e81c3cb8990e27532996fd7. However, this feels wrong and is quite confusing

Reply:
What would you think about calling the class `VOITransformation`? That would be a slight deviation from the standard, but at least it would make clear that it is not a `LUT`.

Reply:
This conversation is now outdated due to changes introduced by https://github.com/herrmannlab/highdicom/pull/166